models/dataset.go

```diff
@@ -11,7 +11,10 @@ import (
 	uuid "github.com/satori/go.uuid"
 )
 
-const created = "created"
+const (
+	created               = "created"
+	editionConfirmedState = "edition-confirmed"
```

Comment:
Consider making the names consistent here. One state is suffixed with 'state' and the other is not.